python/cuml/model_selection/_split.py

```diff
@@ -72,15 +72,19 @@ def _stratify_split(X, y, n_train, n_test, x_numba, y_numba, random_state):
 
     n_classes = classes.shape[0]
     class_counts = cp.bincount(y_indices)
+    if cp.min(class_counts) < 2:
+        raise ValueError("The least populated class in y has only 1"
+                         " member, which is too few. The minimum"
+                         " number of groups for any class cannot"
+                         " be less than 2.")
+
     if n_train < n_classes:
         raise ValueError('The train_size = %d should be greater or '
                          'equal to the number of classes = %d' % (n_train,
                                                                   n_classes))
-    if n_test < n_classes:
-        raise ValueError('The test_size = %d should be greater or '
-                         'equal to the number of classes = %d' % (n_test,
-                                                                  n_classes))
-    class_indices = cp.array_split(cp.argsort(y_indices), n_classes)
+
+    class_indices = cp.split(cp.argsort(y_indices),
+                             cp.cumsum(class_counts)[:-1].tolist())
```

Comment:
Could we add a simple pytest (probably based on the issue being closed)?

Reply:
Yup, added it!